File: Atividade 4 - Haskell/problema15.hs
mdc :: Int -> Int -> Int
mdc m n
	| (m == 0 ) = n
	| (m > 0) = mdc (n `mod` m) m

verificaCoprimo :: Int -> Int
verificaCoprimo 1 = 1
verificaCoprimo _ = 0

totiente :: Int -> Int -> Int
totiente 1 _ = 0
totiente n var = verificaCoprimo(mdc (n-1) var) + (totiente (n-1) var)




main = do
	putStrLn("Digite o primeiro numero: ")
	nStr <- getLine
	let n = (read nStr :: Int)
	let var = (read nStr :: Int)
	print(totiente n var)

File: ProLog/Atividade 15/problema15.pl
% Regra:
mdc(X,Y,Resultado) :- Y > X, mdc(Y,X,Resultado).
mdc(X,Y,Resultado) :- (Y > 0, Z is X mod Y, mdc(Y,Z,Resultado));(Y == 0, Resultado is X).

/*Consultas: 
?- mdc(10,0,X).
X = 10.

?- mdc(30,10,X).
X = 10 ;
false.

?- mdc(250,25,X).
X = 25 ;
false.

?- mdc(250,100,X).
X = 50 ;
false.

*/